pkg/processor/sumologicschemaprocessor/aggregate_attributes_processor.go

```diff
@@ -0,0 +1,255 @@
+// Copyright 2022 Sumo Logic, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicschemaprocessor
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// aggregateAttributesProcessor
+type aggregateAttributesProcessor struct {
+	aggregations []*aggregation
+}
+
+type aggregation struct {
+	attribute      string
+	patternRegexes []*regexp.Regexp
+}
+
+func newAggregateAttributesProcessor(config []aggregationPair) (*aggregateAttributesProcessor, error) {
+	aggregations := []*aggregation{}
+
+	for i := 0; i < len(config); i++ {
+		pair, err := pairToAggregation(&config[i])
+		if err != nil {
+			return nil, err
+		}
+		aggregations = append(aggregations, pair)
+	}
+
+	return &aggregateAttributesProcessor{aggregations: aggregations}, nil
+}
+
+func pairToAggregation(pair *aggregationPair) (*aggregation, error) {
+	regexes := []*regexp.Regexp{}
+
+	for i := 0; i < len(pair.Patterns); i++ {
+		// We do not support regexes - only wildcards (*). Escape all regex special characters.
```

Comment:
why not?

Reply:
We should either support regexes and require a named capture group, or only support prefixes. See: https://github.com/SumoLogic/sumologic-otel-collector/issues/866#issuecomment-1357672421

Reply:
There's a problem with naming in that case. But yes, now that I think about it, we can just force the user to specify at least one capture group